### PR TITLE
LPD-42681 Token definitions are only available for the last theme

### PR DESCRIPTION
**Pull request by izaera**
LPD: https://liferay.atlassian.net/browse/LPD-42681
Sign-off: https://github.com/liferay-frontend/liferay-portal/pull/4578#pullrequestreview-2467086113
QA analysis: :green_circle: https://github.com/liferay-frontend/liferay-portal/pull/4578#issuecomment-2498232891

### CODE REVIEW

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
Re https://github.com/liferay-frontend/liferay-portal/pull/4578#issuecomment-2498007459 I wanted to check if the tests I modified would be run and it turns out that's the case, so good news :tada:

File: modules/apps/frontend-token/frontend-token-definition-impl/src/main/java/com/liferay/frontend/token/definition/internal/FrontendTokenDefinitionRegistryImpl.java [hunk 1]
```diff
@@ -191,35 +200,37 @@ protected String getServletContextName(Bundle bundle) {
...
 			Matcher matcher = _themeIdPattern.matcher(xml);
 
-			if (!matcher.matches()) {
-				return null;
-			}
+			while (matcher.find()) {
+				String themeId = matcher.group(1);
 
-			String themeId = matcher.group(1);
+				String servletContextName = getServletContextName(bundle);
```
Comment (dsanz) [inline modules/apps/frontend-token/frontend-token-definition-impl/src/main/java/com/liferay/frontend/token/definition/internal/FrontendTokenDefinitionRegistryImpl.java:222]: nit: If this is a constant value we can move it outside of the loop

Comment (izaera) [reply modules/apps/frontend-token/frontend-token-definition-impl/src/main/java/com/liferay/frontend/token/definition/internal/FrontendTokenDefinitionRegistryImpl.java:222]: 👍

**Comment by izaera** [comment]
ci:forward

**Pull request closed without merging**